nucleus/deployment/common/src/main/java/fish/payara/deployment/util/GAVConvertor.java

```diff
@@ -135,11 +146,53 @@ private GAVConvertor() {
      * and version number.
      * @return A String representing the relative URI of the provided GAV.
      */
-    private static String constructRelativeURIString(Map<String, String> GAVMap) {
-        final String artefactFileName = GAVMap.get("artefactId") + "-" + GAVMap.get("versionNumber");
+    private static String constructRelativeURIString(Map<String, String> GAVMap, Collection<String> repositoryURIs) {
+        String relativeURI = GAVMap.get("groupId") + "/" + GAVMap.get("artefactId") + "/"
+                + GAVMap.get("versionNumber");
+        String artefactFileName = GAVMap.get("artefactId") + "-" + GAVMap.get("versionNumber");
+        
+        //Check if version is not a snapshot
+        if (!relativeURI.endsWith("SNAPSHOT")) {
+            return relativeURI + "/" + artefactFileName;
+        }
+        //Loop through each repoURI
+        for (String repoURI : repositoryURIs) {
+            URI mavenMetaDataURI = null;
+            try {
+                mavenMetaDataURI = new URI(repoURI + relativeURI + "/" + "maven-metadata.xml");
+                
+                //Check if maven metadata exists
+                if (URIUtils.exists(mavenMetaDataURI)) {
+                    File mavenMetaData = URIUtils.convertToFile(mavenMetaDataURI);
+                    DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+                    documentBuilderFactory.setAttribute(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+                    DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+                    Document document = documentBuilder.parse(mavenMetaData);
+                    document.getDocumentElement().normalize();
+                    NodeList nodeList = document.getElementsByTagName("snapshotVersion");
+                    Node node = nodeList.item(0);
+                    Element element = (Element) node;
+                    //Get latest snapshot version
+                    String version = element.getElementsByTagName("value").item(0).getTextContent();
+                    artefactFileName = GAVMap.get("artefactId") + "-" + version;
+                    logger.log(Level.FINE, "Found version {0} from maven-metadata.xml", version);
+                    break;
+                }
+            } catch (URISyntaxException e) {
+                logger.log(Level.WARNING, "Error creating maven metadata URI");
```

Comment:
this should be combined into a single catch statement i.e. 
```
catch (URISyntaxExceptino | IOExceptino | Parser... e) {}
```

Reply:
That would mean that the log statements would need to be the same.